packages/xo-vmdk-to-vhd/src/virtual-buffer.js

```diff
@@ -1,35 +1,52 @@
 'use strict'
 
-import { Slicer } from 'pipette'
-
 export class VirtualBuffer {
   constructor(readStream) {
-    this.slicer = new Slicer(readStream)
+    const _this = this
+    this.readStream = readStream
+    readStream.on('readable', () => {
+      _this.tryToMoveQueue()
+    })
+    readStream.on('error', error => {
+      this.queue.forEach(e => e.reject(error))
+      this.queue.length = 0
+    })
+    readStream.on('end', () => {
+      this.queue.forEach(e => e.reject())
+      this.queue.length = 0
+    })
+    this.queue = []
     this.position = 0
-    this.promise = null
   }
 
-  get isDepleted() {
-    return !this.slicer.readable
+  tryToMoveQueue() {
```

Comment:
Prefix with `_` every members that are not part of the public API of this class.